Doc/library/asyncio-task.rst

```diff
@@ -103,6 +103,29 @@ To actually run a coroutine, asyncio provides three main mechanisms:
       world
       finished at 17:14:34
 
+* The :class:`asyncio.TaskGroup` class provides a more modern
+  alternative to :func:`create_task`.
+  Using this API the last example becomes::
```

Comment:
I think this is a little backwards: instead of introducing `asyncio.create_task` first and later saying that there is a different recommended alternative, I think we should *start* with `TaskGroup.create_task` and present `asyncio.create_task` as the *unstructured* alternative.

Reply:
I had considered that, but we are not far enough along on the road to universal adoption of TaskGroup to present it that way. And honestly I think it makes sense to first show how you can do something using basic primitives (say, take a group of two rocks and a group of three rocks and count them, finding five rocks) before introducing the higher-level primitives (e.g., roman numerals :-). Once you've seen the TaskGroup version you're never going back to the primitive approach, so the only alternative would be to *replace* the basic version with the TaskGroup example, and that would leave too much other parts of the documentation hanging.

In this PR I am just *introducing* TaskGroup, not rewriting everything that mentions tasks to direct the user to TaskGroup (yet). I do plan to insert references to TaskGroup in other places though.

Reply:
I see, sounds good then.